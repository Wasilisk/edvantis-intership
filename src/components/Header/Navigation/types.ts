import {LinkType} from "../../../models/common";

export type NavigationProps = {
    links: LinkType[]
}