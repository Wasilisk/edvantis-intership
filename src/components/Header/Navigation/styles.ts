import styled from "styled-components";

export const NavigationStyles = styled.ul`
  display: flex;
  justify-content: space-between;
  align-items: center;
  padding: 0;
  margin: 0;
  
  li {
    list-style: none;
    margin: 5px 8px;
    
    a {
      text-decoration: none;
      color: black;
      font-weight: 500;
      transition: color 0.2s;
      
      :hover {
        color: #7bb57c;
      }
    }
  }
`