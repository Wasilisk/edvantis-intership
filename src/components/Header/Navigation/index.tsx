import React from 'react';
import {NavigationStyles} from "./styles";
import {NavigationProps} from "./types";
import {Link} from "react-router-dom";

const Navigation = ({links, isLoading}: NavigationProps) => {
    return (
        <NavigationStyles>
            {
                !isLoading && links && links.map((link, index) => <li
                        key={index}
                    >
                        <Link to={link.to}>{link.name}</Link>
                    </li>)
            }
        </NavigationStyles>
    );
};

export default Navigation;