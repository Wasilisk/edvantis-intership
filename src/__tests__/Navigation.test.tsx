import {render} from "@testing-library/react";
import Navigation from "../components/Header/Navigation";
import {mockLinks} from "./__mock__";

describe('Navigation component', () => {
    it('should render navigation links', () => {
        const {asFragment, getByText} = render(<Navigation isLoading={false} links={mockLinks}/>);
        expect(getByText('Link 2')).toBeInTheDocument();
        expect(asFragment()).toMatchSnapshot();
    })
    it('should be empty while header is loading', () => {
        const {container} = render(<Navigation isLoading={true} links={mockLinks}/>);
        expect(container.firstChild).toBeEmptyDOMElement()
        expect(container).not.toHaveTextContent('Link 1');
    })
})