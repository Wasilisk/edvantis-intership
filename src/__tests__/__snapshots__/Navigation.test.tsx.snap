// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation component should render navigation links 1`] = `
<DocumentFragment>
  <ul
    class="sc-bczRLJ bQPqtL"
  >
    <li>
      <a
        href="/link1"
      >
        Link 1
      </a>
    </li>
    <li>
      <a
        href="/link2"
      >
        Link 2
      </a>
    </li>
    <li>
      <a
        href="/link3"
      >
        Link 3
      </a>
    </li>
    <li>
      <a
        href="/link4"
      >
        Link 4
      </a>
    </li>
  </ul>
</DocumentFragment>
`;
